include/sys/arc_impl.h

```diff
@@ -240,6 +240,9 @@ struct arc_buf_hdr {
 	uint64_t		b_birth;
 
 	arc_buf_contents_t	b_type;
+	uint8_t			b_complevel;
```

Comment:
When reading through, it wasn't clear to me how the levels end up stored in this byte. Is it straight mapped to `enum zio_zstd_levels` ?

Reply:
I think @brainslayer (hinthint) would be the best to tell you where precisely...

Reply:
@lundman yes. this is basicly mapped to zio_zstd_levels. see zstd_cookie_to_enum in zstd.c which translates the zstd level (which can be negativ) this this internal stored value

Reply:
Gotcha, and enums are terrible to try to force a size onto. Cheers.

Reply:
Thanks for your time @lundman  :)

Reply:
@lundman if i'm correct enums are defined as integer but the size is unspecified. so a fixed size of a field would be correct.

Reply:
The reason I didn't use the enum here is that it also maps to the gzip level, and any other future compression algorithm